samples/subsys/tracing/app.overlay

```diff
@@ -0,0 +1,19 @@
+/*
```

Comment:
I guess this should be renamed to something else, i.e. `gpio.overlay`, as `app.overlay` has special meaning and will be added by the build system [by default](https://docs.zephyrproject.org/latest/develop/application/index.html#overview)

Reply:
That shouldn't really be a huge concern though, for a sample, should it? 

Alexander did add app.overlay earlier due to this comment
https://github.com/zephyrproject-rtos/zephyr/pull/74199#discussion_r1697269111

Reply:
> Alexander did add app.overlay earlier due to this comment
> [#74199 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/74199#discussion_r1697269111)

The gpio test is now completely contained in a testcase of its own, so this looks like the final piece of a 1k pieces puzzle to me

> That shouldn't really be a huge concern though, for a sample, should it?

This sample doubles as a test, but yeah, it is not going to break anything so I'm going to stamp it